### PR TITLE
Initial Authentication Modes doc

### PR DESCRIPTION
**Pull request by rosswbrown**
Document the supported auth modes with sample connection dialog, connection resolver.  Also include notes about the hive specific constraints.

### CODE REVIEW

File: docs/auth-modes.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    </connection-resolver>
+    ...
+</tdr>
+```
+
+### Multiple Authentication Modes
+
+User is prompted for which authentication option to use, then a set of fields appear, conditional on that option.  Depending on the option selected the user may or may not be prompted for credentials when reconnecting to the data source.
+
+Support authention options are:
```
Comment (kpwilson) [inline docs/auth-modes.md]: authention = authentication

File: docs/auth-modes.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+            <option name="None" value="auth-none" />
+            <option name="Username" value="username" />
+            <option name="UsernameAndPassword" value="username-password" default="true" />
+        </authentication-options>
+        ...
+    </connection-config>
+</connection-dialog>
+```
+
+Note: the ```value``` attribute value for all options are customizable by connector author, except None, which is required to be ```auth-none```.  These option values are the persisted value of the authentication attribute in a Tableau Workbook (twb) or Tableau Data Source (tds) file.
```
Comment (kpwilson) [inline docs/auth-modes.md]: are -> is  "...attribute value for all options is customizable by ..."  I'd also suggest that you lower-case "workbook" and "data source"

File: docs/auth-modes.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+    ...
+})
+```
+
+## Considerations for 'hadoophive' and 'spark' base classes
+
+When using 'hadoophive' or 'spark' as a base class there are additional constraints and requirements in the Connection Dialog and Connection Resolver.
+
+The Connection Dialog authentication options ```option``` element's ```value``` attribute's value must match below.
```
Comment (kpwilson) [inline docs/auth-modes.md]: I'm having a hard time parsing this sentence -- is it possible to construct the sentence to say "this" must match "this"?

Comment (rosswbrown) [reply docs/auth-modes.md]: I was having trouble with this section as well.  I think its a little better now.

**Pull request merged**